apiclient/harvester_api/managers.py

```diff
@@ -730,3 +737,169 @@ def softreboot(self, name, namespace=DEFAULT_NAMESPACE, *, raw=False):
+    CREATE_PATH_fmt = "v1/harvester/{SC_API}.storageclasses"
+
+    PATH_fmt = "/apis/{SC_API}/v1/storageclasses/{name}"
+
+    def get(self, name="", *, raw=False, **kwargs):
+        path = self.PATH_fmt.format(SC_API=self.API_VERSION, name=name)
+        return self._get(path, raw=raw, **kwargs)
+
+    def create_data(self, name, replicas):
+        data = {
+            "type": "{self.API_VSERION}",
+            "metadata": {
+                "name": name
+            },
+            "parameters": {
+                "numberOfReplicas": f"{replicas}",
+                "staleReplicaTimeout": "30",
+                "migratable": "true"
+            },
+            "provisioner": "driver.longhorn.io",
+            "allowVolumeExpansion": True,
+            "reclaimPolicy": "Delete",
+            "volumeBindingMode": "Immediate"
+        }
+
+        return data
+
+    def create(self, name, replicas=3, *, raw=False):
+        path = self.CREATE_PATH_fmt.format(SC_API=self.API_VERSION)
+        data = self.create_data(name, replicas)
+        return self._create(path, json=data, raw=raw)
+
+    def set_default(self, name, *, raw=False):
+        code, resp = self.get()
+
+        if code == 200 and len(resp['items']):
+            for sc in resp['items']:
+                if sc['metadata'].get('annotations', {}).get(
+                   DEFAULT_STORAGE_CLASS_ANNOTATION) == 'true':
+
+                    if sc['metadata']['name'] == name:
+                        return (200, sc)
+
+                    # reset old default storage class
+                    path = self.PATH_fmt.format(name=sc['metadata']['name'],
+                                                SC_API=self.API_VERSION)
+                    resp = self._patch(path, json={
+                        "metadata": {
+                            "annotations": {
+                                "storageclass.kubernetes.io/is-default-class": "false",
+                                "storageclass.beta.kubernetes.io/is-default-class": "false"
+                            }
+                        }
+                    }, raw=raw)
+
+                    break
+
+        path = self.PATH_fmt.format(name=name, SC_API=self.API_VERSION)
+        return self._patch(path, json={
+            "metadata": {
+                "annotations": {
+                    "storageclass.kubernetes.io/is-default-class": "true",
+                    "storageclass.beta.kubernetes.io/is-default-class": "true"
+                }
+            }
+        }, raw=raw)
+
+    def delete(self, name, *, raw=False):
+        path = self.PATH_fmt.format(name=name, SC_API=self.API_VERSION)
+        return self._delete(path, raw=raw)
+
+
+class VersionManager(BaseManager):
+    PATH_fmt = "apis/harvesterhci.io/v1beta1/namespaces/{namespace}/versions/{name}"
+
+    API_PATH_fmt = "v1/harvester/harvesterhci.io.versions/{namespace}{name}"
+
+    def get(self, name, namespace=DEFAULT_HARVESTER_NAMESPACE, *, raw=False, **kwargs):
```

Comment:
consider that `get` will also need to allow fetch all items, so the `name` should have default value like:
```python
def get(self, name="", namespace=DEFAULT_HARVESTER_NAMESPACE, *, raw=False, **kwargs):
```